kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/DeploymentUtils.java

```diff
@@ -277,4 +278,21 @@ public void conditionEvaluated(EvaluatedCondition<PodStatus> condition) {
             // unused
         }
     }
+
+    /**
+     * Gets node port service url.
+     *
+     * @param namespace the namespace
+     * @param serviceName the service name
+     * @return the node port service url
+     */
+    public static String getNodePortServiceUrl(String namespace, String serviceName) {
+        var nodeIP = kubeClient(namespace).getClient().nodes().list().getItems().get(0).getStatus().getAddresses().get(0).getAddress();
```

Comment:
How about use the Streams API?  It is slightly easier on the eye and should fail more gracefully (with a meaningful error) if the lists are ever unexpectedly empty.

```suggestion
        var nodes = kubeClient(namespace).getClient().nodes().list().getItems();
        var nodeIP = nodes.stream().findFirst()
                .map(Node::getStatus)
                .map(NodeStatus::getAddresses)
                .stream().findFirst()
                .orElseThrow(() -> new KubeClusterException("Unable to get IP of the first node from " + nodes));
```